streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -233,8 +228,13 @@ public void print(final KeyValueMapper<? super K, ? super V, String> mapper,
                       final String label) {
         Objects.requireNonNull(mapper, "mapper can't be null");
         Objects.requireNonNull(label, "label can't be null");
-        String name = builder.newName(PRINTING_NAME);
-        builder.internalTopologyBuilder.addProcessor(name, new KStreamPrint<>(new PrintForeachAction<>(null, mapper, label), keySerde, valSerde), this.name);
+        print(Printed.<K, V>toSysOut().withLabel(label).withKeyValueMapper(mapper));
```

Comment:
This does not seem to be backward compatible as we ignore `keySerde` and `valSerde` now...

Reply:
Yep, there was some discussion about it here: https://github.com/apache/kafka/pull/3636#discussion_r131981250 and also offline with @guozhangwang

Reply:
The issue is that the `key` and `value` are already deserialized into the expected type, i.e., if the stream is `<Integer, String>` then you already have  `Integer key, String value` so the serdes are pointless. You can't pass in another type of `Serde` as they have to be `Serde<K>` and `Serde<V>`

Reply:
Ack. Thanks for clarification.